libarchive/test/test_read_format_rar5.c

```diff
@@ -932,19 +932,19 @@ DEFINE_TEST(test_read_format_rar5_symlink)
 	assertEqualInt(AE_IFLNK, archive_entry_filetype(ae));
 	assertEqualString("file.txt", archive_entry_symlink(ae));
 	assertEqualInt(AE_SYMLINK_TYPE_FILE, archive_entry_symlink_type(ae));
-	assertA(0 == archive_read_data(a, NULL, archive_entry_size(ae)));
+	assertEqualInt(0, archive_entry_size(ae));
```

Comment:
This test should continue to verify that `archive_read_data` returns zero here.  I think the best thing is to keep the line you've added and put back the previous check as a separate line:  that is, verify that `archive_entry_size` returns zero, and then that requesting zero bytes results in a zero byte read.

Reply:
Okay, I have added it again. It requires a cast to size_t though. But since we have checked that it's 0 before, this has no negative impact. Otherwise a simple cast on 32 bit systems could hide a bug if archive_entry_size is larger than SIZE_MAX. Although every other 64 bit system would notice that.